### PR TITLE
etcdserver: export method EtcdServer.leaderChangedNotify (#12378)

### PR DESCRIPTION
**Pull request by Yanhao**

### CODE REVIEW

**Review by ptabor — CHANGES_REQUESTED** [review]
Please justify the change / describe problem you are trying to solve. 
 
From first glance it sounds to me as undesired, as the space is implementation detail of etcd.

**Comment by Yanhao** [comment]
When use etcd in Embed mode，the only way to know etcd leader changes is to poll the EtcdServer.Lead, and that way wastes some cpu cycles.
Why I want to know the current leader of  etcd is because that I have another application level leader and I want the leader can follow etcd's easily so there will less network overhead, most request is made by the application level leader after all.
If leaderChangedNotify is exported, then application code can be notified after etcd leader changes,  avoid wasteful pollings.
@ptabor

File: etcdserver/server.go [hunk 1]
```diff
@@ -1689,7 +1689,7 @@ func (s *EtcdServer) getLead() uint64 {
 	return atomic.LoadUint64(&s.lead)
 }
 
-func (s *EtcdServer) leaderChangedNotify() <-chan struct{} {
+func (s *EtcdServer) LeaderChangedNotify() <-chan struct{} {
```
Comment (ptabor) [inline etcdserver/server.go:1692]: Please add it to the 'Server' interface. 

Please comment on the semantic:
  1. that this channel is being closed when the leadership changes.
  2. so the new channel needs to be obtained for each term. 
  3. That you can loose some consecutive channel changes using this API.

Comment (Yanhao) [reply etcdserver/server.go]: done

**Comment by ptabor** [comment]
Thank you. 

now PASSES="fmt" ./test.sh fails with: 

```
% (cd server && 'go' 'vet' './...')
stderr: # go.etcd.io/etcd/server/v3/etcdserver/api/etcdhttp
stderr: vet: etcdserver/api/etcdhttp/peer_test.go:86:41: cannot use &(fakeServer literal) (value of type *fakeServer) as etcdserver.Server value in argument to newPeerHandler: missing method LeaderChangedNotify
stderr: # go.etcd.io/etcd/server/v3/etcdserver/api/v2http
stderr: vet: etcdserver/api/v2http/client_test.go:670:12: cannot use &(serverRecorder literal) (value of type *serverRecorder) as etcdserver.ServerV2 value in struct literal: missing method LeaderChangedNotify
```

**Comment by Yanhao** [comment]
Fixed

**Comment by ptabor** [comment]
Thank you.

**Pull request merged**